src/core/chart-formatters.js

```diff
@@ -0,0 +1,284 @@
+  var hiddenSeries = [];
+
+  _.each(graph.data.columns, column => {
+    var seriesName = column[0];
+    if(!_.isEqual(seriesName, "x")) {
+      if(predicate(column)){
+        hiddenSeries.push(seriesName);
+      }
+    }
+  });
+
+  if(!graph.legend) {
+    graph.legend = {};
+  }
+
+  graph.legend.hide = hiddenSeries;
+  return graph;
+};
+
+/*
+ * Post-processing graph function that sets the order of the data series.
+ * The last-drawn series is the most clearly visible; its points and lines
+ * will be on top where they intersect with other series.
+ *
+ * Accepts a C3 graph and a ranking function. The ranking function will be
+ * applied to each series in the graph, and the series will be sorted by the
+ * ranking function's results. The higher a series is ranked, the later it
+ * will be drawn and the more prominent it will appear.
+ */
+var sortSeriesByRank = function(graph, ranker) {
+  var sorter = function(a, b) {return ranker(a) - ranker(b);}
+  graph.data.columns = graph.data.columns.sort(sorter);
+  return graph;
+};
+
+/*
+ * Post-processing graph function that hides specific series from the tooltip.
+ * 
+ * Takes a graph specification object and a predicate. Any series for which
+ * the predicate returns true will be blocked from appearing in the tooltip.
+ * 
+ * By default, every series appears in the tooltip. This postprocessor is 
+ * only needed if you want one or more series NOT to be shown.
+ */
+var hideSeriesInTooltip = function(graph, predicate) {
+  //determine which series do not appear in the tooltip
+  let hidden = [];
+  
+  for(let i = 0; i < graph.data.columns.length; i++) {
+    if(predicate(graph.data.columns[i])) {
+      hidden.push(graph.data.columns[i][0]);
+    }
+  }
+
+  //in order to have a value not show up in the tooltip, it needs to
+  //render as undefined in the tooltip value formatting function. 
+  //Return undefined for values in the series list made earlier.
+  const oldTooltipValueFormatter = graph.tooltip.format.value;  
+  const newTooltipValueFormatter = function(value, ratio, id, index) {
+    if(hidden.indexOf(id) !== -1) {
+      return undefined; 
+    }
+    else {
+      return oldTooltipValueFormatter(value, ratio, id, index);
+    }  
+  };
+  graph.tooltip.format.value = newTooltipValueFormatter;
+  return graph;
+}
+
+/*
+ * Post-processing graph function that adds extra space above or below
+ * data on a graph by setting the y-axis maximums and minimums to multiples
+ * of the data span. Especially useful if you have data on both the y1 and y2
+ * axis, but don't want them to visually overlap.
+ * 
+ * Arguments:
+ *   graph - the graph to be modified
+ *   axis - either "y1" or "y2"
+ *   direction - where to add padding, either "top" or "bottom"
+ *   padding - the amount of extra y-axis space to add, expressed as a 
+ *             multiple of the existing data span. 
+ */
+var padYAxis = function (graph, axis = "y", direction = "top", padding = 1) {
```

Comment:
Not sure where this habit of declaring functions as variables got started, but it's unnecessary and in some environments can result in anonymous functions where you really want a named function. (I think Babel is smart about naming functions, but some environments aren't and these kinds of anonymous functions make debugging a lot harder.) 

Much prefer a named function, i.e.,
```
function padYAxis(...) { ... }
```

Reply:
On further reflection, I'm not sure if fixing this issue throughout belongs in this PR. It's always tempting to slip in some code improvement like this, but it does add unnecessary changes.